### ISSUE TITLE
Add HLL++ support for better accuracy and possibly lower memory cost 

### ISSUE THREAD

**Issue by lakshmanan-v** [apache/incubator-pinot#7014]
DISTINCTCOUNTHLL accuracy and memory footprint can be improved through latest HLL algorithms. We have a choice either replace the existing implementation with a better one or leave the existing DISTINCTCOUNTHLL to implement original HLL and create separate functions (ex: DISTINCTCOUNTHLLPLUSPLUS).

[Google's HLL++](http://static.googleusercontent.com/external_content/untrusted_dlcp/research.google.com/en/us/pubs/archive/40671.pdf) -- a popular algorithm amongst the community offers lot of improvements over original HLL. There are multiple java implementations of HLL++. Most of them have variations in performance due to the register size and other implementation choices. Clearspring [stream-lib](https://github.com/addthis/stream-lib) used for current HyperLogLog function, implements HLL++ as [HyperLogPlus](https://github.com/addthis/stream-lib/commits/master/src/main/java/com/clearspring/analytics/stream/cardinality/HyperLogLogPlus.java).

**Comment by Jackie-Jiang** [comment]
Another alternative is the Data Sketch HLL: https://datasketches.apache.org/docs/HLL/HLL.html
They claim better performance in this paper: https://datasketches.apache.org/docs/HLL/Hll_vs_CS_Hllpp.html

**Comment by lakshmanan-v** [comment]
Good point @Jackie-Jiang. We could add this as a separate function DISTINCTCOUNTHLLSKETCH (similar to DISTINCTCOUNTTHETASKETCH). It would be nice to document the parameters, accuracy and space requirements of each of HLL implementations as we have a handful of them already. Users can chose the right one based on their need.
